.github/workflows/test-spyre.yml

```diff
@@ -1,29 +1,119 @@
 name: test-sypre
 
-on: pull_request
+on:
+  # Don't use pull_request.paths filter since this workflow is required for
+  # all pull requests on main irrespective of file type or location
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+    paths:
+      - "tests/**/*.py"
+      - "vllm_spyre/**/*.py"
+      - pyproject.toml
+      - .github/workflows/test-spyre.yml
+  workflow_dispatch:
+
+env:
+  # force output to be colored for non-tty GHA runner shell
+  FORCE_COLOR: "1"
+  # prefer index for torch cpu version and match pip's extra index policy
+  UV_EXTRA_INDEX_URL: "https://download.pytorch.org/whl/cpu"
```

Comment:
Is this required? I see in the torch install step below we use `pip` and set the index url directly